app/_kong_plugins/ai-mcp-proxy/index.md

```diff
@@ -210,11 +220,173 @@ rows:
+
+### How default and per-tool ACLs work
+
+The plugin evaluates access using a two-tier system:
+
+<!-- vale off -->
+{% table %}
+columns:
+  - title: ACL type
+    key: field
+  - title: Description
+    key: description
+rows:
+  - field: |
+      [`default_acl`](./reference/#schema--config-default-acl)
+    description: |
+      Baseline rules that apply to all tools unless overridden.
+  - field: |
+      [`tools[].acl`](./reference/#schema--config-tools-acl)
+    description: |
+      When configured, these rules replace the default ACL for that specific tool. The per-tool ACL doesn't inherit or merge with `default_acl`—it is an all-or-nothing override.
+{% endtable %}
+<!-- vale on -->
+
+{:.info}
+> If a tool defines its own ACL, the plugin ignores `default_acl` for that tool:
+>
+> - Tools with no ACL configuration inherit the default rules (both `allow` and `deny` lists)
+> - Tools with an ACL must explicitly list all allowed subjects (even if they were already in `default_acl`)
+
+### ACL evaluation logic
+
+Both default and per-tool ACLs use `allow` and `deny` lists. Evaluation follows this order:
+
+1. **Deny list**: If the subject matches any `deny` entry, the request is rejected (`INVALID_PARAMS -32602`).
+2. **Allow list (optional)**: If an `allow` list exists, the subject must match at least one entry; otherwise, the request is denied (`INVALID_PARAMS -32602`).
+3. **Only deny configured**: If no `allow` list exists and the subject is not in `deny`, the request is allowed.
+4. **No ACL configuration**: If neither list exists, the request is allowed.
+
+All access attempts (allowed or denied) are written to the plugin's audit log.
+
+The table below summarizes the possible ACL configurations and their outcomes.
+
+{% table %}
+columns:
+  - title: Condition
+    key: condition
+  - title: "Proxied to upstream service?"
+    key: proxy
+  - title: Response code
+    key: response
+rows:
+  - condition: "Subject matches any `deny` rule"
+    proxy: No
+    response: INVALID_PARAMS -32602
+  - condition: "`allow` list exists and subject is not in it"
+    proxy: No
+    response: INVALID_PARAMS -32602
+  - condition: "Only `deny` list exists and subject is not in it"
+    proxy: Yes
+    response: 200
+  - condition: "No ACL rules configured"
+    proxy: Yes
+    response: 200
+{% endtable %}
+
+### ACL tool control request flow
+
+The AI MCP Proxy plugin evaluates ACLs for both tool discovery and tool invocation. These are two distinct operations with different behaviors:
+
+**Tool Discovery (List tools)**:
+1. MCP client requests the list of available tools
+2. Kong AuthN plugin validates the request and identifies the Consumer
+3. AI MCP Proxy loads the Consumer's group memberships
+4. Plugin evaluates the default ACL (`default_acl`) for each tool
```

Comment:
I'm confused about "default ACL for each tool", does it evaluate tool-specific ACLs?